### PR TITLE
a11y(core:accordion): rename core accordion panel to section

### PR DESCRIPTION
**Pull request by hippee-lee**
We have a wrapper container that groups an accordion header and its
content. We call this a 'panel' but this leads to a lot of confusion as
the aria- documentation refers to just the content part as 'panel'.
In this change we rename the container to section to avoid this confusion.

Signed-off-by: Jeeyun Lim <jeeyun.lim@gmail.com>

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
There a `cds-accordion-panel` for wrapping header and content elements.

Issue Number: N/A

## What is the new behavior?
There a `cds-accordion-section` for wrapping header and content elements.

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

This renames one of the cdsAccordion children elements. 
It will impact anyone who uses the core accordion. 
The migration path is to audit codebases for use of `cds-accordion-panel` and replace it with `cds-accordion-section`.

### CODE REVIEW

**Review by coryrylan — APPROVED** [review]

**Review by bbogdanov — APPROVED** [review]

**Review by mathisscott — APPROVED** [review]
I have no issue with the work done here – beyond it being a breaking change with no discernable improvement to the component in question. I feel compelled, however, to speak up about how weird this request is. 

It is an API change, driven by the a11y team. I'm not comfortable with the precedent that all variable, method, and class names now fall under the umbrella of the a11y team. 

The work here is fine. The request behind it feels like overreach.

And I'm not convinced that this is an area where the a11y team has expertise or a long-standing interest in developer experience and API stability.

**Review by gnomeontherun — CHANGES_REQUESTED** [review]
Blocking just for a moment to discuss.

As I view this there are a few concerns.

1. We are changing the API, but we don't yet have a great way to help automate this kind of update for teams.
2. There is nothing about this change that improves features. 
3. Documentation for this is going to have to be managed across the new website work, and we don't have a great way to provide details about version specific changes in the website (and therefore code samples and demos would be wrong for anyone not on the latest).
4. Migration from Angular to Core would be more confusing, and we'd have to update our tooling to properly handle the v5 or v6 API depending on what is installed.

In short, there are side effects (more than I listed) for no feature or bugfix gain. There is also nothing about this change that has to go in right now, but if it goes in we'll have to spawn a series of additional tasks just to handle this consistently. 

Are the side effects of this change worth it?

**Comment by hippee-lee** [comment]
I believe that this change was a request from a11y but I am not 100% sure since Jeeyun drove the code and I have just maintained it since she left.

**Comment by hippee-lee** [comment]
Closing this for now. Context for the original change has been lost and 

1. it a breaking change that needs more support work for good DX (linters/fixes
2. This could be handled with an aliased class in the the future to provide a new element tag.
